client/finality-grandpa/src/environment.rs

```diff
@@ -1177,13 +1181,29 @@ pub(crate) fn finalize_block<BE, Block, Client>(
 						commit,
 					)?;
 
-					Some(justification.encode())
+					Some(justification)
 				} else {
 					None
 				}
 			},
 		};
 
+		// Notify any registered listeners in case we have a justification
+		if let Some(ref justification) = justification {
+			match client.header(BlockId::Hash(hash)) {
+				Ok(Some(header)) => {
+					let notification = (header, justification.clone());
+					if let Some(sender) = justification_sender {
+						let _ = sender.notify(notification);
```

Comment:
I *think* this should be the correct plate to notify, but would be good if the reviewers could give this part extra attention, thanks :)